dojo/tasks.py

```diff
@@ -310,3 +314,45 @@ def async_dupe_delete(*args, **kwargs):
 @task(name='celery_status', ignore_result=False)
 def celery_status():
     return True
+
+
+@app.task(rate_limit=1, ignore_result=True)
+def synchronize_vulnerability_mirrors():
+    def chunk_task(task, file_names, chunk_size):
+        files = list(file_names)
+        count = len(files)
+        file_chunks = (files[i:i + chunk_size] for i in range(0, count, chunk_size))
+        return [task.s(chunk) for chunk in file_chunks]
+
+    # FIXME: this should check the updated time of the state to avoid re-processing in-progress data
+    last_processed_revision = VulnerabilityMirrorState.get_last_processed_revision(settings.VULNDB_URL)
+    mirror = VulnerabilityMirror(logger=logger)
+    checkpoint_revision = mirror.revision
+    header = chunk_task(process_new_vulnerability_files, mirror.added_files(last_processed_revision), 2000)
+    if last_processed_revision is not None and len(last_processed_revision) > 0:
+        header.extend(
+            chunk_task(process_updated_vulnerability_files, mirror.updated_files(last_processed_revision), 2000))
+    chord(header)(checkpoint_vulnerability_state.si(checkpoint_revision))
```

Comment:
I'll note that while the checkpointing won't be done until after the files are processed, if this task is invoked again while the first invocation is still running, you'll end up with two parallel lists of the same files to be imported which will log a lot of uniqueness constraint errors. I'm thinking of using a timestamp in the state model here to ensure that only one task of this type is running at a time regardless if this top level task finished (this task doesn't wait for the remaining tasks to complete, so rate limiting doesn't help much here in practice).